### DISCUSSION TITLE
How can we improve the Store API?

### DISCUSSION THREAD

**Discussion by ralucaStan** [woocommerce/woocommerce#43553] [Checkout Flow]
Since March 2022 the [Store API is considered stable](https://developer.woocommerce.com/2022/03/25/store-api-is-now-considered-stable/). In November 2022 we announced that [The Store API Now Supports Rate Limiting](https://developer.woocommerce.com/2022/11/22/store-api-now-supports-rate-limiting/). You can visit the Store API documentation page for more details.

-----------

This is an open-ended question to gather some feedback from the community about any issues, or missing functionality you have identified. 

We're specifically interested in any issues that may be stopping merchants or developers from adopting the Cart & Checkout blocks over the shortcode version.

**Comment by ant-magic42** [comment]
The Store API can be used to add an item to cart and a Cart-Token JWT header then provides the cart contents to a headless frontend, eg SPA, mobile app, etc. The cart can be retrieved, items updated, etc. So far so good.

But a headless frontend currently has no way to send the customer to the actual checkout  with the cart (via the Cart-Token) and commence a normal checkout session to use the new checkout blocks.

Thus a headless frontend is forced to implement its own checkout and forced to only use the Store API to checkout.

It would be great for a headless frontend to be able to use the checkout blocks instead of having to build a headless checkout.

**Comment by CHEWX** [comment]
Worthy information here — https://x.com/se_nadir/status/1830538873702605124?s=46&t=lKOU2UwWz8Ldx0-9o9bGyg

![IMG_3192](https://github.com/user-attachments/assets/371e3cbb-4bbb-4c17-a13d-f6338a68377a)
